pattern_matcher/pattern_matcher.go

```diff
@@ -0,0 +1,76 @@
+package pattern_matcher
+
+import "github.com/gobwas/glob"
+
+type PatternMatcher struct {
+	include []string
+	exclude []string
+}
+
+// Option is the type for functional options.
+type Option func(*PatternMatcher)
+
+func New(options ...Option) *PatternMatcher {
+	pm := &PatternMatcher{
+		include: []string{"*"},
+		exclude: []string{},
+	}
+
+	for _, option := range options {
+		option(pm)
+	}
+	return pm
+}
+
+func WithInclude(list []string) Option {
+	return func(c *PatternMatcher) {
+		c.include = list
+	}
+}
+
+func WithExclude(list []string) Option {
+	return func(c *PatternMatcher) {
+		c.exclude = list
+	}
+}
+
+func WithIncludeFilter(value string) Option {
```

Comment:
WithInclude(values ...string) which appends the string(s) to an existing arrayt of filters.